### PR TITLE
KYLIN-4396 Close FileReader in SaveDictStep

### PR DESCRIPTION
**Pull request by hit-lacus**

### CODE REVIEW

File: engine-mr/src/main/java/org/apache/kylin/engine/mr/streaming/SaveDictStep.java [hunk 1]
```diff
@@ -136,6 +136,7 @@ public boolean accept(Path path) {
                         logger.error("dictionary of column {} not found! ", colRef.getName());
                     }
                 }
+                reader.close();
```
Comment (shaofengshi) [inline engine-mr/src/main/java/org/apache/kylin/engine/mr/streaming/SaveDictStep.java]: Will it be more safe to put it into a "finally" block?

Comment (hit-lacus) [reply engine-mr/src/main/java/org/apache/kylin/engine/mr/streaming/SaveDictStep.java]: Thank you!

File: engine-mr/src/main/java/org/apache/kylin/engine/mr/streaming/SaveDictStep.java [hunk 2]
```diff
@@ -108,32 +108,32 @@ public boolean accept(Path path) {
...
-                            new ByteArrayInputStream(dictInfo.getBytes())));
-
-                    Dictionary dict = dictionaryInfo.getDictionaryObject();
-                    if (dict != null) {
-                        dictionaryInfo = dictManager.trySaveNewDict(dict, dictionaryInfo);
-                        cubeSeg.putDictResPath(colRef, dictionaryInfo.getResourcePath());
-                        if (cubeSeg.getRowkeyStats() != null) {
-                            cubeSeg.getRowkeyStats().add(
-                                    new Object[] { colRef.getName(), dict.getSize(), dict.getSizeOfId() });
+                try(SequenceFile.Reader reader = new SequenceFile.Reader(fs, file.getPath(), conf)) {
```
Comment (shaofengshi) [inline engine-mr/src/main/java/org/apache/kylin/engine/mr/streaming/SaveDictStep.java]: The code format here is not well. Please format it...

Comment (hit-lacus) [reply engine-mr/src/main/java/org/apache/kylin/engine/mr/streaming/SaveDictStep.java]: Thank you, I have updated.

**Pull request merged**